File: backend/app/schemas/parking.py
# app/schemas/parking.py
from pydantic import BaseModel, Field
from datetime import datetime
from enum import Enum
from typing import Optional

class VehicleType(str, Enum):
    CAR = "CAR"
    MOTORCYCLE = "MOTORCYCLE"
    TRUCK = "TRUCK"

class ParkingSpotBase(BaseModel):
    spotNumber: str = Field(alias='spot_number')
    vehicleType: VehicleType = Field(alias='vehicle_type')
    occupied: bool = False

    class Config:
        from_attributes = True
        populate_by_name = True

class ParkingSpot(ParkingSpotBase):
    id: int

class VehicleBase(BaseModel):
    licensePlate: str = Field(alias='license_plate')
    vehicleType: VehicleType = Field(alias='vehicle_type')

    class Config:
        from_attributes = True
        populate_by_name = True

class Vehicle(VehicleBase):
    id: int
    entryTime: datetime = Field(alias='entry_time')
    exitTime: Optional[datetime] = Field(default=None, alias='exit_time')
    parkingSpotId: int = Field(alias='parking_spot_id')

class ParkingResponse(BaseModel):
    licensePlate: str = Field(alias='license_plate')
    spotNumber: str = Field(alias='spot_number')
    entryTime: datetime = Field(alias='entry_time')
    message: str

    class Config:
        from_attributes = True
        populate_by_name = True

File: frontend/src/types/parking.ts
// src/types/parking.ts
export enum VehicleType {
    CAR = "CAR",
    MOTORCYCLE = "MOTORCYCLE",
    TRUCK = "TRUCK"
  }
  
  export interface Vehicle {
    id: number;
    license_plate: string;
    vehicle_type: VehicleType;
    entry_time: string;
    exit_time: string | null;
    parking_spot_id: number;
  }
  
  export interface ParkingSpot {
    id: number;
    spot_number: string;
    vehicle_type: VehicleType;
    occupied: boolean;
  }
  
  export interface ParkingResponse {
    license_plate: string;
    spot_number: string;
    entry_time: string;
    message: string;
  }
  
  // Additional types for requests if needed
  export interface ParkVehicleRequest {
    licensePlate: string;
    vehicleType: VehicleType;
  }
  
  export interface ExitVehicleRequest {
    licensePlate: string;
  }